### DISCUSSION TITLE
Eric Elliott - SudoLang

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#203] [Announcements]
- http://devanywhere.io/
- https://twitter.com/_ericelliott?lang=en
- https://twitter.com/SudoLang

## Questions

- [x] What the state of AI coding tools? 
- [x] What is sudolang?
- [x] Is there a spec? Or does psuedo code really mean that an i can write stuff that looks like code?
- [x] How would I go about learning sudolang?
- [x] How do I teach an LLM sudolang?
- [x] Does it work with any LLM?
- [x] How does the psuedo code lead to better results?
- [x] How is a program represented in sudolang? Do I define functions?
- [x] Do I approach coding like this in a fundamentally different way?
- [x] Can I extend sudolang with my own syntax?
- [x] Is it indempotent? Can I test it?
- [x] Do you think this has use cases for "production"? or is it more of a personal tool? How would I host it?
- [x] What are commands? Why would I want to use them?
- [x] What are constraints?
- [x] What types of programs are suited for this type of programming? What isn't?

### Future

- [x] Do you think this will become the predominant way of coding? Or do you think more new ways will emerge?

**Comment by ericelliott** [comment]
Looks great. Please be sure to spell my name Eric Elliott with two t's. Thanks!

**Reply by just-be-dev** [reply]
Apologies for that! Got it fixed.

**Comment by hipstersmoothie** [comment]
https://www.cursor.so

**Comment by ericelliott** [comment]
* VS Code
* GitHub Co-Pilot
* ChatGPT Advanced Data Analysis (GPT-4 + Python Interpreter)
* DailyBot https://www.dailybot.com/
* SonarCloud https://sonarcloud.io/

**Comment by just-be-dev** [comment]
Tooltips: 
- https://github.com/binwiederhier/ntfy
- https://docs.astro.build/en/guides/view-transitions/
